auth/service.py

```diff
@@ -101,17 +103,23 @@ def authorize_google(self, request: Request, code: str) -> Response:
 
       state = request.cookies.get("oauth_state")
       if state is not None:
-          target_path = base64.urlsafe_b64decode(
-              state.removeprefix("b'").removesuffix("'")
-          ).decode("utf-8")
+        cleaned_state = state.removeprefix("b'").removesuffix("'")
+        target_path = base64.urlsafe_b64decode(
+          cleaned_state
+        ).decode("utf-8")
 
       target_url = self.base_url + target_path
       response = RedirectResponse(url=target_url)
-      response.set_cookie(key="token", value=jwt_token, httponly=True, secure=True)
+      response.set_cookie(
+        key="token", 
+        value=jwt_token, 
+        httponly=True, 
+        secure=True,
+      )
       
       return response
     
-    def get_user_profile(self, token: str) -> ProfileResponse:
+    def get_user_profile(self, token: str) -> UserModel:
```

Comment:
Can you explain why the change from `ProfileResponse` to `UserModel` was done? The previous approach of decoupling the database model from the representation in the domain was already good

Reply:
Precisely the opposite of that, because we need to decouple the Controller (handling responses) and the Service (handling just logic). I think the `AuthService` can be more versatile in future, because the service should not care what the HTTP Response looks like.

Reply:
If that's the intention, it would be better to use the `User` class from `user.py` instead of the one tied to the repository

Reply:
Can I ask why? The repository returns a `UserModel`, would it not naturally make sense to use `UserModel` ?

Reply:
The `UserModel` only makes sense within the context of the repository. It's how it interfaces with the underlying database. However, the consumer is the service which doesn't care about that and only how it behaves, which is why the `User` class is better.

It's not really a pattern exclusive to pure backend. FastAPI actually endorses and supports this in their tutorial pages with Pydantic